internal/helm/helmdriver/transport_http.go

```diff
@@ -23,54 +23,194 @@ import (
 	kitxhttp "github.com/sagikazarmark/kitx/transport/http"
 
 	"github.com/banzaicloud/pipeline/.gen/pipeline/pipeline"
+	"github.com/banzaicloud/pipeline/internal/clustergroup/deployment"
 	"github.com/banzaicloud/pipeline/internal/helm"
 	apphttp "github.com/banzaicloud/pipeline/internal/platform/appkit/transport/http"
+	helm2 "github.com/banzaicloud/pipeline/pkg/helm"
 )
 
 func RegisterHTTPHandlers(endpoints Endpoints, router *mux.Router, options ...kithttp.ServerOption) {
 	errorEncoder := kitxhttp.NewJSONProblemErrorResponseEncoder(apphttp.NewDefaultProblemConverter())
 
-	router.Methods(http.MethodPost).Path("").Handler(kithttp.NewServer(
+	router.Methods(http.MethodPost).Path("/repos").Handler(kithttp.NewServer(
 		endpoints.AddRepository,
 		decodeAddRepositoryHTTPRequest,
 		kitxhttp.ErrorResponseEncoder(kitxhttp.StatusCodeResponseEncoder(http.StatusAccepted), errorEncoder),
 		options...,
 	))
 
-	router.Methods(http.MethodGet).Path("").Handler(kithttp.NewServer(
+	router.Methods(http.MethodGet).Path("/repos").Handler(kithttp.NewServer(
 		endpoints.ListRepositories,
 		decodeListRepositoriesHTTPRequest,
 		kitxhttp.ErrorResponseEncoder(encodeListRepositoriesHTTPResponse, errorEncoder),
 		options...,
 	))
 
-	router.Methods(http.MethodDelete).Path("/{name}").Handler(kithttp.NewServer(
+	router.Methods(http.MethodDelete).Path("/repos/{name}").Handler(kithttp.NewServer(
 		endpoints.DeleteRepository,
 		decodeDeleteRepositoryHTTPRequest,
 		kitxhttp.ErrorResponseEncoder(encodeDeleteRepositoryHTTPResponse, errorEncoder),
 		options...,
 	))
 
-	router.Methods(http.MethodPatch).Path("/{name}").Handler(kithttp.NewServer(
+	router.Methods(http.MethodPatch).Path("/repos/{name}").Handler(kithttp.NewServer(
 		endpoints.PatchRepository,
 		decodePatchRepositoryHTTPRequest,
 		kitxhttp.ErrorResponseEncoder(kitxhttp.StatusCodeResponseEncoder(http.StatusAccepted), errorEncoder),
 		options...,
 	))
 
-	router.Methods(http.MethodPut).Path("/{name}").Handler(kithttp.NewServer(
+	router.Methods(http.MethodPut).Path("/repos/{name}").Handler(kithttp.NewServer(
 		endpoints.UpdateRepository,
 		decodeUpdateRepositoryHTTPRequest,
 		kitxhttp.ErrorResponseEncoder(kitxhttp.StatusCodeResponseEncoder(http.StatusAccepted), errorEncoder),
 		options...,
 	))
+
+	router.Methods(http.MethodGet).Path("/charts").Handler(kithttp.NewServer(
+		endpoints.ListCharts,
+		decodeListChartsHTTPRequest,
+		kitxhttp.ErrorResponseEncoder(encodeListChartsHTTPResponse, errorEncoder),
+		options...,
+	))
+
+	// TODO fix the path after migrating to h3 (use chartS instead of chart) - backwards  compatibility!
+	router.Methods(http.MethodGet).Path("/chart/{reponame}/{name}").Handler(kithttp.NewServer(
+		endpoints.GetChart,
+		decodeChartDetailsHTTPRequest,
+		kitxhttp.ErrorResponseEncoder(encodeChartDetailsHTTPResponse, errorEncoder),
+		options...,
+	))
+}
+
+func RegisterReleaserHTTPHandlers(endpoints Endpoints, router *mux.Router, options ...kithttp.ServerOption) {
```

Comment:
IMHO this should eventually be moved under `/helm/releases` and the deployment API should be retired.

Reply:
thx, will do it later / i wouldn't break the api at this point